### DISCUSSION TITLE
How can I translate the position

### DISCUSSION THREAD

**Discussion by jacobxy** [saul/demofile#210] [Questions] [answered]
How can I convert the position in the file to the position in the map ?
The postion I get is like this (3305171700,1159083300, 1094367000) . 
I don't know where is the point in the map of the match !

**Comment by saul** [comment]
What do you mean by "position in the map"? If you're referring to the place name (e.g "SiteB") you can get that from the m_szLastPlaceName network property.

**Comment by jacobxy** [comment]
For example, in the horizontal and vertical coordinates of the map !  
By these values, we can generally know where the player is on the map ！

**Comment by saul** [comment]
When you say "map" do you mean an X,Y co-ordinate on the overview image? Like these:

![image](https://cloud.githubusercontent.com/assets/64654/24836494/30043e22-1d15-11e7-9998-7655794b0d21.png)

**Comment by jacobxy** [comment]
Yes， it's the map! 
I want to know where is the point with the position such as (3305171700,1159083300, 1094367000) .

**Comment by saul** [comment, marked as answer]
Take a look at these .json files: https://github.com/saul/gamevis/tree/master/overviews/csgo

Imagine you have a world-space co-ordinate named `worldSpace`, such as {x: 3305.17170, y: 11590.83300}.

To determine where this position fits on the map, do:

```f#
map_x = (worldSpace.x - pos_x) / scale
map_y = (worldSpace.y - pos_y) / scale
```

Where pos_x, pos_y and scale are taken from: https://github.com/saul/gamevis/blob/master/overviews/csgo/de_dust2.json

**Comment by jacobxy** [comment]
Thanks a lot ! 
I konw what can I do next! 
: )

**Comment by Brammz** [comment]
What do these `pos_x` and `pos_y` mean exactly?

I am trying to draw locations of a match on de_cache using `<svg>` from 0 to 768.
The minimum x and y values are -1395 and -1447 respectively and the maximum x and y values are 2214 and 2261 respectively.
So for the [de_cache.json](https://github.com/saul/gamevis/blob/master/overviews/csgo/de_cache.json) file the `map_x` and `map_y` values can be both positive and negative.

I am trying to use it in the the following way:

```javascript
let circles = [];
for (let pos of positions) {
  circles.push(
    <circle cx={map_x} cy={map_y} r="5" fill="red" />
  )
}

<svg viewBox="0 0 768 768">
  <image href={image} alt="map" height="768" width="768" />
  {circles}
</svg>
```

**Comment by saul** [comment]
@Brammz, you want to set `cx` and `cy` to:

```
let circles = [];
for (let pos of positions) {
  circles.push(
    <circle cx={(pos.x - pos_x) / scale} cy={(pos.y - pos_y) / scale} r="5" fill="red" />
  )
}
```

For de_cache, `pos_x = -2278`, `pos_y = 3469` and `scale = 5.5`.

**Comment by Brammz** [comment]
Thanks, it works now for kills and deaths! However for any type of grenade it gives x, y and z values between 0 and 32 everytime. (Just getting the entity position on detonate event)

**Comment by saul** [comment]
I've just released 0.4.16 which fixes this

**Comment by Brammz** [comment]
It works now, thank you so much! You're awesome!

**Comment by agpmilli** [comment]
Hi, I am working on a custom map, it means that I cannot use the .json you linked above for the coordinate translation. Do you have an idea how I can get the custom map pos_x, pos_y and scale ?
Thanks!

Edit: More info: When I am on my map and I use "getpos" it outputs  
`-1363.000000 1094.050049 65.093811` as my position  
but at the same moment in the demo I obtain  
`3299500032 1149813120 1065615360`

What should I use to do the mapping from one to the other?

Even if I can get pos_x, pos_y and scale and use these formulas to find my map_x and map_y:  
`map_x = (worldSpace.x - pos_x) / scale`  
`map_y = (worldSpace.y - pos_y) / scale`  
I still have some trouble understanding how you can use them to obtain **Z**, **pitch**, **yaw** and **roll** in function of the map scaling and position

**Comment by saul** [comment]
I assume this custom map has an overview script in `resource/overviews/map_name_here.txt`? You can find pos_x, pos_y and scale from there.

However, if this is a map that you have made yourself, you can create a mini-map by following this Valve Dev Wiki article: [Creating a working mini-map](https://developer.valvesoftware.com/wiki/Creating_a_working_mini-map)

**Comment by agpmilli** [comment]
Thanks for your answer. Yes I made this map myself so I don't have any `resource/overviews/my_map.txt` and I don't see how creating a minimap can help me go back to the "getpos" coordinate I obtain in the game. Do you have an idea ?

**Comment by saul** [comment]
Can you raise a separate issue about your position problem? It’s not really related to this. Add the demo file you’re referring to if possible.
